engine/access/state_stream/event.go

```diff
@@ -0,0 +1,56 @@
+package state_stream
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type ParsedEventType int
+
+const (
+	ProtocolEventType ParsedEventType = iota + 1
+	AccountEventType
+)
+
+type ParsedEvent struct {
+	Type      ParsedEventType
+	EventType flow.EventType
+	Address   string
+	Contract  string
+	Name      string
+}
+
+// ParseEvent parses an event type into its parts. There are 2 valid EventType formats:
+// - flow.[EventName]
+// - A.[Address].[Contract].[EventName]
+// Any other format results in an error.
+func ParseEvent(eventType flow.EventType) (*ParsedEvent, error) {
+	parts := strings.Split(string(eventType), ".")
+
+	switch parts[0] {
+	case "flow":
+		if len(parts) == 2 {
+			return &ParsedEvent{
+				Type:      ProtocolEventType,
+				EventType: eventType,
+				Contract:  parts[0],
```

Comment:
Someone could define a contract named `flow` with the same event names which would break anyone filtering on the contract `flow`.